actors/user/src/main/java/org/sunbird/user/actors/util/AuthenticationHelper.java

```diff
@@ -0,0 +1,169 @@
+package org.sunbird.user.actors.util;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.models.util.StringFormatter;
+import org.sunbird.common.models.util.datasecurity.EncryptionService;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.util.Util;
+
+public class AuthenticationHelper {
+
+  private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private static EncryptionService encryptionService =
+      org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.getEncryptionServiceInstance(
+          null);
+
+  public static void validateWithUserId(Request reqObj) {
+    if (null != reqObj.getContext().get(JsonKey.IS_AUTH_REQ)
+        && Boolean.parseBoolean((String) reqObj.getContext().get(JsonKey.IS_AUTH_REQ))) {
+      String ctxtUserId = (String) reqObj.getContext().get(JsonKey.USER_ID);
+      String userId = getUserIdFromExtIdAndProvider(reqObj);
+      if ((!StringUtils.isBlank(userId)) && (!userId.equals(ctxtUserId))) {
+        throw new ProjectCommonException(
+            ResponseCode.unAuthorized.getErrorCode(),
+            ResponseCode.unAuthorized.getErrorMessage(),
+            ResponseCode.UNAUTHORIZED.getResponseCode());
+      }
+    }
+  }
+
+  private static String getUserIdFromExtIdAndProvider(Request reqObj) {
```

Comment:
Add this method in UserService.getUserId(Request request)

Reply:
validateWithUserId - used from within this method - so not added to service layer.